js/main.js

```diff
@@ -32,3 +32,115 @@ function submitUserInput(event) {
+      </li>
+    </div>
+  </div>
+</ul>
+*/
+
+function addEntry(entries) {
+
+  var newUnorderedList = document.createElement('ul');
+  var dataViewEntry = document.querySelector('div[data-view="entries"]');
+  dataViewEntry.appendChild(newUnorderedList);
+
+  var divRow = document.createElement('div');
+  divRow.setAttribute('class', 'row');
+  newUnorderedList.appendChild(divRow);
+
+  var divColumnHalf1 = document.createElement('div');
+  divColumnHalf1.setAttribute('class', 'column-half');
+  divRow.appendChild(divColumnHalf1);
+
+  var liImage = document.createElement('li');
+  divColumnHalf1.appendChild(liImage);
+
+  var image = document.createElement('img');
+  image.setAttribute('src', entries.url);
+  image.setAttribute('class', 'dom-tree-image');
+  liImage.appendChild(image);
+
+  var divColumnHalf2 = document.createElement('div');
+  divColumnHalf2.setAttribute('class', 'column-half');
+  divRow.appendChild(divColumnHalf2);
+
+  var liText = document.createElement('li');
+  divColumnHalf2.appendChild(liText);
+
+  var heading = document.createElement('h2');
+  heading.textContent = entries.title;
+  liText.appendChild(heading);
+
+  var bodyText = document.createElement('p');
+  bodyText.textContent = entries.notes;
+  liText.appendChild(bodyText);
+
+  return newUnorderedList;
+}
+
+window.addEventListener('DOMContentLoaded', function appendEntries() {
+  for (var i = 0; i < data.entries.length; i++) {
+    var dataViewEntries = document.querySelector('div[data-view="entries"]');
+    dataViewEntries.appendChild(addEntry(data.entries[i]));
+  }
+});
+
+document.addEventListener('submit', function appendSubmission() {
+  var dataViewEntries = document.querySelector('ul');
+  var hideForm = document.getElementById('entry-form');
+  dataViewEntries.prepend(addEntry(data.entries[0]));
+  hideForm.classList.add('hidden');
+  var showEntryView = document.getElementById('entries-view');
+  showEntryView.classList.remove('hidden');
+  var hideNoEntryText = document.getElementById('no-entries');
+  hideNoEntryText.classList.add('hidden');
+  location.href = '#entries-view';
+});
+
+var newEntryButton = document.querySelector('.new-entry');
+
+newEntryButton.addEventListener('click', function showForm() {
+  var entryForm = document.getElementById('entry-form');
+  entryForm.classList.remove('hidden');
+  var hideEntryView = document.getElementById('entries-view');
+  hideEntryView.classList.add('hidden');
+});
+
+var navBarEntries = document.getElementById('entries-nav');
+navBarEntries.addEventListener('click', function hideForm() {
+  var entryForm = document.getElementById('entry-form');
+  entryForm.className = 'container view hidden';
+  var entryView = document.getElementById('entries-view');
+  entryView.className = 'container view';
+});
```

Comment:
Once you fix how you are setting `data.view`, these functions should be able to be removed in favor of just using `assignClassNames` (which again should be renamed to something like `switchView`)